pennylane_lightning/src/util/Kokkos_Sparse.hpp

```diff
@@ -185,16 +185,21 @@ namespace Pennylane::Util {
  */
 template <class fp_precision, class index_type>
 void apply_Sparse_Matrix_Kokkos(
-    [[maybe_unused]] const std::complex<fp_precision> *vector_ptr,
-    [[maybe_unused]] const index_type vector_size,
-    [[maybe_unused]] const index_type *row_map_ptr,
+    const std::complex<fp_precision> *vector_ptr, const index_type vector_size,
+    const index_type *row_map_ptr,
     [[maybe_unused]] const index_type row_map_size,
-    [[maybe_unused]] const index_type *entries_ptr,
-    [[maybe_unused]] const std::complex<fp_precision> *values_ptr,
+    const index_type *entries_ptr, const std::complex<fp_precision> *values_ptr,
     [[maybe_unused]] const index_type numNNZ,
-    [[maybe_unused]] std::vector<std::complex<fp_precision>> &result) {
-    PL_ABORT("Executing the product of a Sparse matrix and a vector needs "
-             "Kokkos and Kokkos Kernels installation.");
+    std::vector<std::complex<fp_precision>> &result) {
+    result.resize(vector_size);
+    size_t count = 0;
+    for (index_type i = 0; i < vector_size; ++i) {
```

Comment:
We always use post-increment in Lightning, unless pre is required for functionality.